### ISSUE TITLE
Inference on Custom DB

### ISSUE THREAD

**Issue by jkstyle2** [ch3cook-fdu/Vote2Cap-DETR#21]
Dear authors,

I'd like to test your model on my custom 3D reconstructed pointclouds + rgb map.
In order to directly use the [checkpoints](https://huggingface.co/CH3COOK/Vote2Cap-DETR-weights/tree/main) you shared, I need to extract the normalization features from 3D mesh (as your models are all trained with the norm features).
However, I have no idea how to make the norm features from my custom dataset. I understand the normalized features are calculated in [read_mesh_vertices_rgb_normal()](https://github.com/ch3cook-fdu/Vote2Cap-DETR/blob/master/data/scannet/scannet_utils.py#L114) from the face of 3D mesh. But I only have 3D pointclouds and rgb, without the mesh faces.
I initially tried to get the faces using Meshlab tool (surface reconstruction - ball pivoting), and the result is as below.
![image](https://github.com/ch3cook-fdu/Vote2Cap-DETR/assets/34189274/f16fdd39-000a-40c5-8b73-7a419fc7442d)

Using the 3D mesh, I followed the all preprocessing steps (xyz, rgb, norm) to extract the similar features to fit your model with the checkpoint(`scanrefer_scst_vote2cap_detr_pp_XYZ_MULTIVIEW_NORMAL.pth`).
But the detection and captioning result was really poor, and I suspect the normalized features would be the issue.

Regarding this issue, can you share you insight?

**Comment by ch3cook-fdu** [comment]
You can train a model with only xyz+rgb inputs and test it on your data.

**Comment by jkstyle2** [comment]
I'm training the model with only xyz+rgb inputs as suggested.
Can you share the training log or any ablation study with/without normal features, if conducted? 
I'd like to know the impact of normal features.

**Comment by jkstyle2** [comment]
The evaluation result from the model trained with only xyz+rgb inputs is as below.
I think it pretty much good without the normal features as well.

----------------------Evaluation-----------------------
INFO: iou@0.5 matched proposals: [1516 / 2068], 
[BLEU-1] Mean: 0.6582, Max: 1.0000, Min: 0.0000
[BLEU-2] Mean: 0.5636, Max: 1.0000, Min: 0.0000
[BLEU-3] Mean: 0.4638, Max: 1.0000, Min: 0.0000
[BLEU-4] Mean: 0.3785, Max: 1.0000, Min: 0.0000
[CIDEr] Mean: 0.7329, Max: 6.3784, Min: 0.0000
[ROUGE-L] Mean: 0.5413, Max: 1.0000, Min: 0.1015
[METEOR] Mean: 0.2608, Max: 1.0000, Min: 0.0448


One thing I'm wondering is that sometimes there is a difference between the semantic class from detection head and the described class from caption head. One of the results is as below. Can you tell me the reasons and how to make them match? 
It seems the caption branch is not using the output of the detection branch when inferencing captions. 

"detection": "sofa"
"caption": "this is a brown table . it is to the left of the table ."

**Comment by ch3cook-fdu** [comment]
Yes, the result seems ok to me. If you would like to alleviate the hallucinations, you can design the alignment as a reward function to supervise the caption generation training.

**Comment by jkstyle2** [comment]
> you can design the alignment as a reward function

Do you mean to design the reinforced learning?

One more thing I'm wondering is that the model currently predicts the color and shape info sometimes, not always.
Can I always get the color and shape attributes as the captioning output?
Do you have any idea of it?
